deeplake/core/vectorstore/deepmemory_vectorstore.py

```diff
@@ -0,0 +1,61 @@
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import numpy as np
+
+from deeplake.core.dataset import Dataset
+from deeplake.core.vectorstore.deeplake_vectorstore import VectorStore
+from deeplake.core.vectorstore.deep_memory import DeepMemory
+from deeplake.constants import DEFAULT_VECTORSTORE_DISTANCE_METRIC
+
+
+class DeepMemoryVectorStore(VectorStore):
+    def __init__(self, *arg, **kwargs):
+        super().__init__(*arg, **kwargs)
+        try:
+            self.deep_memory = DeepMemory(
+                self.dataset,
+                token=self.token,
+                embedding_function=self.embedding_function,
+            )
+        except Exception as e:
+            self.deep_memory = None
+
+    def search(
+        self,
+        embedding_data: Optional[Union[str, List[str]]] = None,
+        embedding_function: Optional[Callable[..., Any]] = None,
+        embedding: Optional[Union[List[float], np.ndarray]] = None,
+        k: int = 4,
+        distance_metric: Optional[str] = None,
+        query: Optional[str] = None,
+        filter: Optional[Union[Dict, Callable[..., Any]]] = None,
+        exec_option: Optional[str] = None,
+        embedding_tensor: str = "embedding",
+        return_tensors: Optional[List[str]] = None,
+        return_view: bool = False,
+        deep_memory=True,
+    ) -> Union[Dict, Dataset]:
+        if exec_option is not None and exec_option != "tensor_db":
+            self.logger.warning(
+                "Specifying `exec_option` is not supported for this dataset. "
+                "The search will be executed on the Deep Lake Managed Database."
+            )
+
+        distance_metric = distance_metric or DEFAULT_VECTORSTORE_DISTANCE_METRIC
```

Comment:
@adolkhan I'm 95% sure this shouldn't be here. The Vector Store search method has a bunch of logic to respect the supported distance metrics from the indexes in the dataset, and by adding this, you're essentially skipping that logic and introducing potential problems. Just pass the `distance_metric` directly to super().search and let that method handle it.

Reply:
There gonna be inconsistency between index and deep_memory, index requires the distance metric it was instantiated with while deep_memory requires the distance metric it was trained with.

Reply:
@khustup said we don't support index + deep memory right now

Reply:
@adolkhan Ok, then we need to add separate error handling for that scenario. 

Have you tried loading a dataset with an index and running search in the Deep Memory API above? I'm pretty sure you will see a bunch of weird behavior where either you see an error, or the distance metric automatically gets changes to the one that's in the index. Either way, the behavior will not be what the user expects. There should be clear error messaging on what's supported, and whether any other search features that the user might expect are being overridden.